daemon/src/projection.rs

```diff
@@ -481,6 +500,37 @@ impl Cfd {
             }
         }
     }
+
+    fn derive_actions(&self, role: Role) -> HashSet<CfdAction> {
+        match (self.state, role) {
+            (CfdState::PendingSetup, Role::Maker) => {
+                HashSet::from([CfdAction::AcceptOrder, CfdAction::RejectOrder])
+            }
+            (CfdState::PendingSetup, Role::Taker) => HashSet::new(),
+            (CfdState::ContractSetup, _) => HashSet::new(),
+            (CfdState::Rejected, _) => HashSet::new(),
+            (CfdState::PendingOpen, _) => HashSet::new(),
+            (CfdState::Open, _) => HashSet::from([CfdAction::Commit, CfdAction::Settle]),
```

Comment:
👍 It looks like you fixed a bug here: we had no actions after rolling over (or failing to roll over) and returning to `CfdState::Open`.